### PR TITLE
fix(passthrough): close a capped streaming turn as truncated, not as an error

### PR DESCRIPTION
**Pull request by materemias**
## The bug

A passthrough turn runs on a one-turn budget, so `max_turns` is its ordinary terminal state, not a fault.

When it trips with tool calls captured, the streaming path already recovers them as `stop_reason: "tool_use"`. When it trips with **none** captured, the turn fell through to the error envelope - and the client had already rendered the turn's text over SSE. So the user saw a finished answer with `Reached maximum number of turns (1)` stamped on it.

141 such terminations in 24h on one machine, all `reason=max_turns turns=1`, all unrecovered:

```
0405add2 sdk_termination reason=max_turns turns=1 model=opus[1m] resume=true deferred=false
```

## The fix

The non-streaming path has degraded honestly since the budget dropped to one: content and no forwardable call is reported as `max_tokens` (`server.ts`, `passthrough.capped_turn_truncated`). Its comment already claimed the streaming path did the same. This makes that true.

`max_tokens` is the choice on both paths for the same reason: it is the wire's word for a cut-off turn, so a client can continue from what it has. An error frame is a dead end, and `end_turn` is the silent-turn lie #768 exists to prevent.

Nothing durable moves. With no captured calls there is no checkpoint to publish and no mapping to advance - exactly as on the throwing path.

## What stays on the error path

Two shapes, each with a test:

- **Only `message_start` forwarded.** Nothing was delivered, so there is nothing to truncate. `eventsForwarded` cannot express this: it counts the envelope's own opening frame, so it is already 1 whenever `messageStartEmitted` is. The gate is `nextClientBlockIndex`, which counts only content blocks the client received - what the non-streaming branch means by `contentBlocks`.
- **A `tool_use` block on the wire with nothing captured.** That means the hook refused those calls (forced-single overflow, duplicate abort, early-stop reversion). Closing with `max_tokens` would dangle a call the client is told neither to run nor to drop.

## Verification

`src/__tests__/passthrough-early-stop-integration.test.ts` gains three cases. The truncation case fails on the parent commit; each boundary case fails when its own clause is removed from the gate. `tsc --noEmit` clean; full suite clean.

### CODE REVIEW

File: src/proxy/server.ts [hunk 1]
```diff
@@ -6065,6 +6065,126 @@ export function createProxyServer(config: Partial<ProxyConfig> = {}): ProxyServe
...
+                nextClientBlockIndex > 0
+              ) {
+                flushOpenClientBlocks("capped_turn")
+                diagnosticLog.session(
+                  `${requestMeta.requestId} sdk_termination_truncated ${formatSdkTermination(sdkTerm, {
+                    model,
+                    requestSource,
+                    isResume,
+                    hasDeferredTools,
+                    sdkSessionId: resumeSessionId,
```
[the remark this answers is not in the document]
Comment (materemias) [reply src/proxy/server.ts]: Fixed in c437a6e: both capped-stream telemetry sites now use `currentSessionId || resumeSessionId`, matching the normal stream success path and preserving correlation for fresh requests. I also added a regression assertion for streamed result usage. Verification: `bun test src/__tests__/passthrough-early-stop-integration.test.ts` — 46 pass; `bun run typecheck` — clean.

**Comment by rynfar** [comment]
Your `test` job is red for a reason that has nothing to do with this PR — please rebase onto `main` and it should clear.

The 11 failures here (`Retry-After (#901)` ×5, `Empty messages array (#450)`, `Passthrough tool_use` ×5) are #917, which was diagnosed and fixed earlier today in #935. The transcript ownership ceiling is bounded per session root, and the test preload pointed all ~179 files at a single root — so prepared transcripts accumulated until it filled, after which every request in the process returned `500 session transcript ownership backlog is full` regardless of which file issued it. That is why unrelated tests fail with impossible statuses, like a 429 test receiving 500.

This branch predates that fix, so it is still running against the broken harness. `git rebase origin/main` and push, and the cluster should disappear.

Apologies for the delay in getting you CI signal on this — the approval was held while the flake made every result untrustworthy.
